### PR TITLE
Blog app integration specs n+1

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?

In this project, I will create integration tests for all of the views used in my project. 

Description of the task to be completed 👎

in this PR I will do bellow tasks 👎

- [x] Solving N+1 problem.

- Before:
N+1 quiry 1
N+1 quiry 2

- After:
N+1 quiry 3
N+1 quiry 4

- [x] Using Capybara to writing integration tests for each view in my project.

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
+require 'rails_helper'
+# rubocop:disable Layout/LineLength
```
Comment (rustam-tolipov) [inline spec/features/post_index_spec.rb]: **Well-written code!**

- [x] I can see you guys put such efford to this project. However, there are some places that need to be fixed up. For starting, in your spef** folder, I can see `rubocop:disable.. `. Using such thing are not allowed in this project. Please remove all of them. 

> HINT: you can create helper folders and put extra codes into it to reduce the size of these files. 
> If these errors are not able to fix you can reach me out, or you can leave these like that. But in my opinion they are fixable :)

File: spec/features/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+RSpec.describe 'Showing all posts', type: :feature do
```
Comment (rustam-tolipov) [inline spec/features/post_index_spec.rb:3]: - [x] You have some failing tests. Due to this reason I was not able to check your code as requested. Please fix them up :)

> COMMANDS I'm using to test: 
> 1. rails db:reset => to clena up the database
> 2. rspec spec => to test 

Screenshot 2023-06-01 at 19 36 49

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    expect(page).to have_content('Page 1')
+  end
+
+  scenario 'When I click on a post, it redirects me to that post show page.' do
+    visit users_path
+    expect(page).to have_content(@first_user.name)
+    click_on @first_user.name
+    visit user_path(@first_user.id)
+    expect(page).to have_content(@first_user.name)
+    # click_on @first_post.title
```
Comment (rustam-tolipov) [inline spec/features/post_index_spec.rb]: - [x] Kindly remove unused commented codes from your file :)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @mrkamin, and @Feliverse ,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Well-implemented code_
_Great readme_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (mrkamin) [reply spec/features/post_index_spec.rb]: @Rustamxon7 
thanks for your good and valuable feedback. 
I apply your feedback and remove it.

File: spec/features/post_index_spec.rb (hunk 2 above)
Comment (mrkamin) [reply spec/features/post_index_spec.rb:3]: @Rustamxon7 
thanks for your good and valuable feedback.
now the test passed.
test passd

File: spec/features/post_index_spec.rb (hunk 3 above)
Comment (mrkamin) [reply spec/features/post_index_spec.rb]: @Rustamxon7 
thanks for your good and valuable feedback.
it removed.

**Comment by Kamba56** [comment]
Hi @mrkamin,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly fix all linters errors.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._
@Kamba56 
thanks for your good and valuable feedback

**Review by descholar-ceo — APPROVED** [review]
Hi @mrkamin 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all test cases ✔️ 
👍🏽 The PR is professional ✔️ 
👍🏽 Linters are passing ✔️ 
👍🏽 Correct usage of git flow ✔️ 

## Optional suggestions
- [x] Just recommending you to install github actions for test cases

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
@descholar-ceo 
thanks for your good and valuable feedback. 
your recommendation applied.

**Pull request merged**